R/get-oa-data.R

```diff
@@ -0,0 +1,77 @@
+# Get open access data from the Unpaywall API
+
+source(here::here("R", "environment.R"))
+
+# Filter trials with a publication and DOI and keep only unique DOIs
+intovalue_dois <- intovalue %>%
+  filter(has_publication, has_doi) %>%
+  distinct(doi, .keep_all = TRUE)
+
+#------------------------------------------------------------------------------#
+# 1. Query Unpaywall API with journal > repository hierarchy (except Bronze)
+#------------------------------------------------------------------------------#
+
+cols <- c("doi","color","issn","journal","publisher","date")
+df <- data.frame(matrix(ncol = length(cols), nrow = 0))
+colnames(df) <- cols
+
+doi_batch <- intovalue_dois[["doi"]]
+print(paste0("DOI number: ", length(doi_batch)))
+
+unpaywall_results <- unpaywallR::dois_OA_colors(doi_batch,
+                                                email_api,
+                                                clusters = 2,
+                                                color_hierarchy = c("gold",
+                                                                    "hybrid",
+                                                                    "green",
+                                                                    "bronze",
+                                                                    "closed"))
+
+oa_results <- tibble(doi = doi_batch,
+                     color = unpaywall_results$OA_color,
+                     issn = unpaywall_results$issn,
+                     journal = unpaywall_results$journal,
+                     publisher = unpaywall_results$publisher,
+                     date = unpaywall_results$date)
+
+df <- rbind(df, oa_results)
+
+all_results <- left_join(intovalue, df, by = "doi")
+
+test <- all_results %>%
+  verify(nrow(.)==nrow(intovalue))
+
+#------------------------------------------------------------------------------#
+# 2. Query Unpaywall API with all OA routes > green OA hierarchy
```

Comment:
@delwen
I'm finding 3 dois with unpaywall results in the green but not the main hierarchy. any idea why this might be?
10.1192/bjp.bp.116.184168
10.1111/jdv.13305
10.1038/leu.2013.7